### DISCUSSION TITLE
Eagle Eye would be far more interesting if you learned spells IN the combat they were used

### DISCUSSION THREAD

**Discussion by Granfalegion** [ihhub/fheroes2#7966] [Ideas]
Eagle Eye is typically very low on anyone's list of desired secondary abilities.  You only benefit from it if (a) your hero has Eagle Eye, (b) your opponent casts a spell you don't have, (c) your RNG rolls well, and (d) your hero also wins the combat.  Only then do you reap any benefit from having the skill.

I just had the idea that EE could be much more interesting if it took effect as soon as an opposing spell was cast, allowing you to cast the learned spell in the very same combat where you had just learned it.

This would introduce an interesting strategic element for otherwise-dominant heroes: "Should I cast my most powerful offensive spell in this fight that I expect I'll otherwise win easily?"  You might hold off on casting Chain Lightning if you think it could force you to face one coming right back at you that you wouldn't otherwise encounter.  This might also encourage heroes that know they're focusing on lightning (spells, Rod) to also pack the Lightning Helm because they're at greater risk of facing combats that feature opposing lightning too.

**Comment by zenseii** [comment]
I think this could create an interesting dynamic and a change for what many consider the worst skill. I would suggest to make it into an artifact that would modify the Eagle Eye skill to also work in combats, this would preserve the original way this skill works.

**Reply by LeHerosInconnu** [reply]
Hello @zenseii,

I'm afraid Eye Eagle would not be more attractive this way, as the possession of the artifact becomes a prerequisite for the secondary skill.
Unless the hero already possesses the artifact, Eye Eagle remains not very desirable.

The artifact would also have a small percentage chance of being present in a scenario, one artifact among many.
Also, the artifact would occupy a slot to make the secondary skill work (to make the skill interesting).

I also prefer, for the "robustness" of the game, that secondary skills be autonomous in the way they function initially.
An artifact can improve a secondary skill, but not correct it so that it works in an attractive way.

**Comment by LeHerosInconnu** [comment]
Hello @Granfalegion,

> I just had the idea that EE could be much more interesting if it took effect as soon as an opposing spell was cast, allowing you to cast the learned spell in the very same combat where you had just learned it.

I couldn't agree more.
And I go one step further for the Eagle Eye secondary skill, see here: https://github.com/ihhub/fheroes2/discussions/2990. :)

**Comment by ihhub** [comment]
The game mustn't have everything ideal. Not every skill, spell, artifact or monster should be the best of the best. The game even has artifacts that are absolutely bad and it is fine. Unlike newer entries in HoMM series Heroes 2 is a game of compromises and it would be even more interesting to play when more "bad" items exist in the game rather than you have to choose between good and good.

**Reply by LeHerosInconnu** [reply]
Hello @ihhub,

When all the skills are good, there's a considered decision to be made about which to pick.
Will I prefer to increase the hero's movement on the adventure map with the logistics skill, or will I prefer the ability to cast higher level spells with the wisdom skill?
It's an interesting choice to make for the player, and it brings "value" to the game.

When a good skill and a bad skill are offered, there's no longer any thoughtful choice to be made.
The player takes the good skill because the other isn't interesting enough.
This becomes an automatism for the player, there is no "satisfaction" for the player in this action, it is not "positive" for the game.

**Reply by Branikolog** [reply]
On the other hand with the current skill and artifact dispersion there are cases, when player has to choose an average skill, instead of the one (the best one) he's looking for. Sometimes player even has to choose between two awful choices, but this brings the diversity into gaming process and forces player to use those skills. I have a lot of fun, when I realize myself using and getting benefits from the skills or artifacts, that are considered as useless. :) I believe that allowing player to build a prefect hero each time he starts a new game makes it boring in perspective. Playing with a hero with not ideal build is a nice challenge. :)
My personal opinion.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> ...has to choose between two awful choices...
> ...forces player to use those skills...

All is said. :)

> I have a lot of fun, when I realize myself using and getting benefits from the skills or artifacts, that are considered as useless. :)

Interesting.
But it can also be seen as a positive reaction to a not so "positive" choice imposed on the player.
In a standard scenario, other than one that would have been specifically designated by its creator to try to showcase the Eagle Eye secondary skill, have you ever thought of wanting to obtain the Eagle Eye skill for one of your heroes to implement a game strategy?
"I'm going to play this scenario this way, I absolutely need my main hero to learn the Eagle Eye secondary skill."

> I believe that allowing player to build a prefect hero each time he starts a new game makes it boring in perspective. Playing with a hero with not ideal build is a nice challenge. :)

The player is never guaranteed a perfect hero, because there are more secondary skills than the hero can learn, even if all the secondary skills are super interesting.
It would be an even more interesting challenge if all secondary skills were super interesting, and the player had to make choices with a strong impact on the game.
